### PR TITLE
OCD-4125 - Determine strategy for refreshing Listing Search cache

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/CertifiedProductManager.java [hunk 1]
```diff
@@ -355,10 +356,10 @@ public CertifiedProductDTO changeOwnership(Long certifiedProductId, Long acbId)
     })
     @CacheEvict(value = {
             CacheNames.ALL_DEVELOPERS, CacheNames.ALL_DEVELOPERS_INCLUDING_DELETED,
-            CacheNames.COLLECTIONS_LISTINGS, CacheNames.COLLECTIONS_SEARCH,
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/CertifiedProductManager.java:358]: Do we still need the COLLECTIONS_LISTINGS here? I thought you would only remove COLLECTIONS_SEARCH

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/developer/MergeDeveloperJob.java [hunk 2]
```diff
@@ -134,6 +134,7 @@ private void clearCachesRelatedToDevelopers() {
         CacheManager.getInstance().getCache(CacheNames.ALL_DEVELOPERS_INCLUDING_DELETED).removeAll();
         CacheManager.getInstance().getCache(CacheNames.COLLECTIONS_DEVELOPERS).removeAll();
         CacheManager.getInstance().getCache(CacheNames.COLLECTIONS_LISTINGS).removeAll();
+        //TODO:  Need to figure this out
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/developer/MergeDeveloperJob.java]: What do we need to do here?

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-resources/src/main/resources/jobs.xml [hunk 3]
```diff
@@ -890,5 +890,14 @@
             <recover>false</recover>
         </job>
         
+        <job>
```
Comment (kekey1) [inline chpl/chpl-resources/src/main/resources/jobs.xml:827]: One more thing - the PR shows a conflict in this file. Requesting changes to address that.

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/ReplaceListingSearchCacheJob.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+        LOGGER.info("********* Starting the Replace Listing Search Cache job. *********");
+        LOGGER.info("Starting retreival of all listings");
+        List<ListingSearchResult> allListings = listingSearchManager.getAllListingsFromDb();
+        LOGGER.info("Completed retreival of all listings");
+        LOGGER.info("Acquiring lock on cache");
+        CacheManager.getInstance().getCache(CacheNames.COLLECTIONS_SEARCH).acquireWriteLockOnKey (CacheNames.COLLECTIONS_SEARCH);
+        LOGGER.info("Repalceing cache");
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/ReplaceListingSearchCacheJob.java]: Noticed this typo in the logs

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Comment by tmy1313** [comment]
Strategy did not work based on our cache replication on the servers.

**Pull request closed without merging**